### ISSUE TITLE
argparse: Check if stderr is defined before writing to it

### ISSUE THREAD

**Issue by simaoafonso-pwt** [python/cpython#101640]
# Bug report

`argparse` tries to write to stderr inconditionally, even when it's `None`. This is only a problem on Windows. Here:

https://github.com/python/cpython/blob/3.11/Lib/argparse.py#L2596-L2600

I detected this when using [PyInstaller](https://pyinstaller.org/en/stable/usage.html) to create a binary on Windows using `--noconsole`, which triggers using `pythonw.exe`. In this case, `sys.stderr is None`.

This is similar to #89057

# Your environment

- Any version of Python on Windows, up to 3.11

### Linked PRs
* gh-101802
* gh-104250

**Comment by terryjreedy** [comment]
@simaoafonso-pwt The title of the 3.11 backport says specifically what was done.  Without a stacktrace, we cannot be sure that this was the site of the error.  And even if it was, whether this is sufficient to fix this issue for you.   If you can, please make a new binary and see if it now works.  And report here so we can either close or look further (given a traceback and message).

**Comment by simaoafonso-pwt** [comment]
> The prebuilt executable will be availible as CPython 3.12b1 on May 22.

I tried to check this, but it seems PyInstaller is not yet 3.12-ready, it might take a longer time: https://github.com/python/cpython/issues/97850

**Comment by terryjreedy** [comment]
I am closing for now.  This can be reopened or a new issue opened if later found necessary.

**Comment by terryjreedy** [comment]
Compiling on Windows is pretty easy now if one has git installed.  Clone python/cpython.  Install VS 2022 with python extension exactly as detailed in devguide.  Give command line, I believe `PCbuild\build.bat -e`.

**Comment by simaoafonso-pwt** [comment]
This is already closed, but only now I got time to check this, now that PyInstaller published wheels for 3.12, and can confirm this is fixed in `3.12b1`. Thanks.

**Comment by simaoafonso-pwt** [comment]
FYI, in case you need a workaround for this issue on lower Python versions, you can use this:

```python
# Workaround for https://github.com/python/cpython/issues/101640
if sys.version_info.minor >= 12:
    ArgumentParser = argparse.ArgumentParser
else:
    class ArgumentParser(argparse.ArgumentParser):
        def _print_message(self, message, file=None):
            if file is None:
                pass  # Don't print if the file is unavailable
            else:
                super()._print_message(message, file=file)

```

Then use `ArgumentParser` instead.
